c++/src/kj/common-test.c++

```diff
@@ -285,9 +285,40 @@ TEST(Common, Maybe) {
     CopyOrMove x(123);
     Maybe<CopyOrMove&> m(x);
     Maybe<CopyOrMove> m2 = kj::mv(m);
-    KJ_EXPECT(KJ_ASSERT_NONNULL(m).i == 123);
+    KJ_EXPECT(m == nullptr);  // m is still moved out of and cleared
```

Comment:
Not sure if I'm understanding this test -- I would expect `m` to be `nullptr` here, but it's a bit weird considering the comment says "Test that initializing Maybe<T> from Maybe<T&>&& does a copy, not a move."

Reply:
Hmm... it looks like what it really meant to test was:

    KJ_EXPECT(x.i == 123);

That is, even though `m` was moved, the thing it pointed to, `x`, was copied, not moved.

Reply:
See #805

Reply:
> That is, even though m was moved, the thing it pointed to, x, was copied, not moved.

Ahhh, got it. But even if a move was performed, wouldn't `x.i` remain 123? I think this is verified from the other end,

https://github.com/capnproto/capnproto/blob/7a31fa889d13286f02a19444e7073d12a6e773c7/c%2B%2B/src/kj/common-test.c%2B%2B#L53-L60

Because if a move was performed, the next line would `== 122`?

Reply:
Wow uh I think that move constructor also has a bug. It should be setting `other.i = -1`. Hence you can tell if the value was moved away because it would be overwritten with -1.

Must have been drunk when I wrote this.

Reply:
Heh, I also kept misreading it as `-= 1`.

Fixed, I think.